src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -1854,8 +1872,17 @@ private RegisteredUser registerUserWithFederatedProvider(final AuthenticationPro
      * @return true if it meets the internal storage requirements, false if not.
      */
     private boolean isUserValid(final RegisteredUser userToValidate) {
-        if (userToValidate.getEmail() == null || userToValidate.getEmail().isEmpty()
-                || !userToValidate.getEmail().matches(".*(@.+\\.[^.]+|-(facebook|google|twitter)$)")) {
+        if (null == userToValidate.getEmail()) return false;
```

Comment:
two returns here. could do it with a single &&?

return null==userToValidate.getEmail() && isEmailValid~